recurly/base_client.py

```diff
@@ -29,7 +29,7 @@ def __init__(self, api_key):
         self.__api_key = api_key
         self.__conn = http.client.HTTPSConnection(HOST, PORT)
 
-    def _make_request(self, method, path, body, params):
+    def _make_request(self, method, path, body, **options):
         try:
             basic_auth = b64encode(bytes(self.__api_key + ":", "ascii")).decode("ascii")
             headers = {
```

Comment:
I've been trying to set these "private" headers after the custom headers to prevent the user from accidentally modifying them.

Reply:
Making a ticket to follow up with that. as well as add validation for options